### PR TITLE
GCS:Uploader: Fix crash with unknown board

### PR DESCRIPTION
**Pull request by tracernz**
Fixes http://dronin-autotown.appspot.com/at/crash/ahFzfmRyb25pbi1hdXRvdG93bnIWCxIJQ3Jhc2hEYXRhGICAgKCXwOEKDA

### CODE REVIEW

File: ground/gcs/src/plugins/coreplugin/boardmanager.cpp [hunk 1]
```diff
@@ -118,6 +118,16 @@ void BoardManager::aboutToRemoveObject(QObject *obj)
         m_boardTypesList.removeAt(m_boardTypesList.indexOf(board));
 }
 
+IBoardType *BoardManager::getBoard(int type)
```
Comment (mlyle) [inline ground/gcs/src/plugins/coreplugin/boardmanager.cpp:121]: Nice factoring.  I think there's other places that do a similar thing we can fix to use this ultimately.

**Comment by tracernz** [comment]
Tested by deleting a board plugin and then plugging it in.

File: ground/gcs/src/plugins/coreplugin/boardmanager.cpp (hunk 1 above)
Comment (tracernz) [inline ground/gcs/src/plugins/coreplugin/boardmanager.cpp:121]: Yep. Gotta run to work though so no time to do it. 👍

**Comment by tracernz** [comment]
Yay, first fix from reading the new style crash reports. :tada:

**Pull request merged**